test/extended/operators/check_crd_status.go

```diff
@@ -0,0 +1,127 @@
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	apiextensionsclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// checkCRDs returns a list of names of CRDs that don't have a "status" in the CRD schema
+// and also a subresource.status defined.
+// For now, it ignores the ones that are currently failing.
+func checkCRDs(crdItemList []apiextensionsv1.CustomResourceDefinition) []string {
+
+	// These CRDs, at the time this test was written, do not have a "status" in the CRD schema
+	// and subresource.status.
+	// These can be skipped for now but we don't want the number to increase.
+	// These CRDs should be tidied up over time.
+	//
+	exceptionsList := sets.NewString(
+		"builds.config.openshift.io",
+		"clusternetworks.network.openshift.io",
+		"consoleclidownloads.console.openshift.io",
+		"consoleexternalloglinks.console.openshift.io",
+		"consolelinks.console.openshift.io",
+		"consolenotifications.console.openshift.io",
+		"consoleplugins.console.openshift.io",
+		"consolequickstarts.console.openshift.io",
+		"consoleyamlsamples.console.openshift.io",
+		"egressnetworkpolicies.network.openshift.io",
+		"hostsubnets.network.openshift.io",
+		"imagecontentpolicies.config.openshift.io",
+		"imagecontentsourcepolicies.operator.openshift.io",
+		"machineconfigs.machineconfiguration.openshift.io",
+		"netnamespaces.network.openshift.io",
+		"networks.config.openshift.io",
+		"networks.operator.openshift.io",
+		"operatorpkis.network.operator.openshift.io",
+		"profiles.tuned.openshift.io",
+		"rangeallocations.security.internal.openshift.io",
+		"rolebindingrestrictions.authorization.openshift.io",
+		"securitycontextconstraints.security.openshift.io",
+		"tuneds.tuned.openshift.io",
+	)
+
+	failures := []string{}
+	for _, crdItem := range crdItemList {
+
+		// This test is interested only in CRDs that end with "openshift.io".
+		if !strings.HasSuffix(crdItem.ObjectMeta.Name, "openshift.io") {
+			continue
+		}
+
+		crdName := crdItem.ObjectMeta.Name
+
+		// Skip CRDs in the exceptions list for now.
+		if exceptionsList.Has(crdName) {
+			continue
+		}
+
+		// Iterate through all versions of the CustomResourceDefinition Spec looking for one with
+		// a schema status element,
+		foundStatusInSchema := false
+		var i int
+		for i = 0; i < len(crdItem.Spec.Versions); i++ {
+			if _, ok := crdItem.Spec.Versions[i].Schema.OpenAPIV3Schema.Properties["status"]; ok {
+				foundStatusInSchema = true
+				break
+			}
+		}
+
+		if foundStatusInSchema {
+			if !(crdItem.Spec.Versions[i].Subresources != nil && crdItem.Spec.Versions[i].Subresources.Status != nil) {
+				failures = append(failures, fmt.Sprintf("CRD %s has a status in the schema but no subresource.status", crdName))
+			}
+		} else {
+			failures = append(failures, fmt.Sprintf("CRD %s has no 'status' element in its schema", crdName))
```

Comment:
ah, this isn't necessarily a bug.

Reply:
Agreed.  But I thought the purpose of this Jira/PR was to check for the presence of the status element in the schema.

Reply:
> Agreed. But I thought the purpose of this Jira/PR was to check for the presence of the status element in the schema.

We have two problems
1. types with status *must* have a subresource.  We've missed some in the past
2. most types should have status.

The second is a lot less severe than the first.  I suggest enforcement by separate tests and this PR just about perfect for 1 which is a lot more important.